### PR TITLE
Notes context feature

### LINKED ISSUE #23: Create notes database context
**Guidelines**
- implement a service interface with available methods (`CreateNote`, `GetAllNotes`, `UpdateNote`, `DeleteNote`)
- add required models and DTO's (`Note`, `NoteDto`)
- add migration due to model modification
- see #5 for inspiration

### PR DESCRIPTION
**Pull request by c1rcle**
Fixes #23.

### CODE REVIEW

File: YellowNotes.Core/Repositories/NoteRepository.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+        public async Task<IEnumerable<Note>> GetNotes(int count, string email,
+            CancellationToken cancellationToken)
+        {
+            return await context.Notes.Where(x => x.UserEmail == email && x.IsRemoved == false)
+                .OrderByDescending(x => x.ModificationDate)
+                .Take(count)
+                .AsNoTracking()
+                .ToListAsync(cancellationToken);
+        }
```
Comment (wkud) [inline YellowNotes.Core/Repositories/NoteRepository.cs:36]: Does this function always returns first couple of notes?

File: YellowNotes.Core/Services/UserService.cs [hunk 2]
```diff
@@ -63,16 +63,5 @@ public string GenerateJWT(UserDto user)
...
-        public bool ValidateToken(string token, UserDto user)
-        {
-            var securityToken = new JwtSecurityTokenHandler().ReadToken(token) as JwtSecurityToken;
-            var decodedEmail = securityToken.Payload["email"] as string;
-
-            var isUserAuthorized = decodedEmail == user.Email;
-            var tokenExpired = securityToken.ValidTo < DateTime.UtcNow;
-
-            return isUserAuthorized && !tokenExpired;
-        }
```
Comment (wkud) [inline YellowNotes.Core/Services/UserService.cs:76]: This function is necessary. Without this validation it's possible to send one email in body and another encoded in token and api will accept such request while it shouldn't.

File: YellowNotes.Core/Repositories/NoteRepository.cs (hunk 1 above)
Comment (c1rcle) [reply YellowNotes.Core/Repositories/NoteRepository.cs:36]: It returns `count` notes or whole collection when `Notes.Count < count`.

File: YellowNotes.Core/Services/UserService.cs (hunk 2 above)
Comment (c1rcle) [reply YellowNotes.Core/Services/UserService.cs:76]: Alright. I'm gonna bring it back from the dead 😃.

File: YellowNotes.Core/Repositories/NoteRepository.cs (hunk 1 above)
Comment (wkud) [reply YellowNotes.Core/Repositories/NoteRepository.cs:36]: Ok, but what happens when user sends `n` requests to get all notes. Will he get whole collection or maybe just first `count` of notes but duplicated `n` times?

Comment (c1rcle) [reply YellowNotes.Core/Repositories/NoteRepository.cs:36]: The notes will be duplicated. I don't think it's a good idea to preserve the loading state between requests. I understand where you're coming from though. I'll think of a better solution.

Comment (wkud) [reply YellowNotes.Core/Repositories/NoteRepository.cs:36]: Some `startIndex` parameter would be cool.

**Review by wkud — APPROVED** [review]

**Review by filip-wan — APPROVED** [review]

**Pull request merged**